### ISSUE TITLE
JSON does not convert without an error

### ISSUE THREAD

**Issue by otichy** [1nVitr0/plugin-vscode-format-list#7]
Hi!

The following valid JSON fails to convert, in fact even the conversion options are not displayed - after selecting List Tools... nothing happens. I tried removing the emoji but that does not seem to be the problem.

```
[{
    "created_at": "Thu Sep 01 10:27:00 +0000 2022",
    "id": 1565285083732496400,
    "id_str": "1565285083732496384",
    "full_text": "WATCH 📺@RepWaltzPress slams Biden’s desperate effort to change the narrative on Democrats’ ‘Defund the Police’ movement and America’s crime crisis:\n\n“I can point to actions all day long that the Left has taken to try and defund our police at every level.” https://t.co/8zjf3WnHt6",
    "truncated": false,
    "display_text_range": [
        0,
        255
    ],
    "entities": {
        "hashtags": [],
        "symbols": [],
        "user_mentions": [
            {
                "screen_name": "RepWaltzPress",
                "name": "Congressman Waltz Press",
                "id": 1082416697380913200,
                "id_str": "1082416697380913152",
                "indices": [
                    7,
                    21
                ]
            }
        ],
        "urls": [],
        "media": [
            {
                "id": 1564980601953943600,
                "id_str": "1564980601953943552",
                "indices": [
                    256,
                    279
                ],
                "media_url": "http://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg",
                "media_url_https": "https://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg",
                "url": "https://t.co/8zjf3WnHt6",
                "display_url": "pic.twitter.com/8zjf3WnHt6",
                "expanded_url": "https://twitter.com/HouseGOP/status/1565285083732496384/video/1",
                "type": "photo",
                "sizes": {
                    "thumb": {
                        "w": 150,
                        "h": 150,
                        "resize": "crop"
                    },
                    "medium": {
                        "w": 1200,
                        "h": 675,
                        "resize": "fit"
                    },
                    "small": {
                        "w": 680,
                        "h": 383,
                        "resize": "fit"
                    },
                    "large": {
                        "w": 1280,
                        "h": 720,
                        "resize": "fit"
                    }
                }
            }
        ]
    },
    "extended_entities": {
        "media": [
            {
                "id": 1564980601953943600,
                "id_str": "1564980601953943552",
                "indices": [
                    256,
                    279
                ],
                "media_url": "http://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg",
                "media_url_https": "https://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg",
                "url": "https://t.co/8zjf3WnHt6",
                "display_url": "pic.twitter.com/8zjf3WnHt6",
                "expanded_url": "https://twitter.com/HouseGOP/status/1565285083732496384/video/1",
                "type": "video",
                "sizes": {
                    "thumb": {
                        "w": 150,
                        "h": 150,
                        "resize": "crop"
                    },
                    "medium": {
                        "w": 1200,
                        "h": 675,
                        "resize": "fit"
                    },
                    "small": {
                        "w": 680,
                        "h": 383,
                        "resize": "fit"
                    },
                    "large": {
                        "w": 1280,
                        "h": 720,
                        "resize": "fit"
                    }
                },
                "video_info": {
                    "aspect_ratio": [
                        16,
                        9
                    ],
                    "duration_millis": 134401,
                    "variants": [
                        {
                            "bitrate": 832000,
                            "content_type": "video/mp4",
                            "url": "https://video.twimg.com/amplify_video/1564980601953943552/vid/640x360/pM47QMwLx9_eCrDB.mp4?tag=14"
                        },
                        {
                            "bitrate": 2176000,
                            "content_type": "video/mp4",
                            "url": "https://video.twimg.com/amplify_video/1564980601953943552/vid/1280x720/OpWFIA3y2UnlWJy5.mp4?tag=14"
                        },
                        {
                            "bitrate": 288000,
                            "content_type": "video/mp4",
                            "url": "https://video.twimg.com/amplify_video/1564980601953943552/vid/480x270/_61qh2kjP6iPER7v.mp4?tag=14"
                        },
                        {
                            "content_type": "application/x-mpegURL",
                            "url": "https://video.twimg.com/amplify_video/1564980601953943552/pl/3XiKmwB_IDNPVHqV.m3u8?tag=14&container=fmp4"
                        }
                    ]
                },
                "additional_media_info": {
                    "title": "Rep. Michael Waltz on Biden's 'Defund the Police' blame game",
                    "description": "",
                    "call_to_actions": {
                        "visit_site": {
                            "url": "https://www.gop.gov/"
                        }
                    },
                    "embeddable": true,
                    "monetizable": false
                }
            }
        ]
    },
    "source": "<a href=\"https://studio.twitter.com\" rel=\"nofollow\">Twitter Media Studio</a>",
    "in_reply_to_status_id": null,
    "in_reply_to_status_id_str": null,
    "in_reply_to_user_id": null,
    "in_reply_to_user_id_str": null,
    "in_reply_to_screen_name": null,
    "user": {
        "id": 15207668,
        "id_str": "15207668",
        "name": "House Republicans",
        "screen_name": "HouseGOP",
        "location": "Washington, DC",
        "description": "We are the Republican Conference in the United States House of Representatives 🇺🇸",
        "url": "https://t.co/lcTnaxp5EE",
        "entities": {
            "url": {
                "urls": [
                    {
                        "url": "https://t.co/lcTnaxp5EE",
                        "expanded_url": "http://gop.gov",
                        "display_url": "gop.gov",
                        "indices": [
                            0,
                            23
                        ]
                    }
                ]
            },
            "description": {
                "urls": []
            }
        },
        "protected": false,
        "followers_count": 1358518,
        "friends_count": 798,
        "listed_count": 7894,
        "created_at": "Mon Jun 23 14:41:52 +0000 2008",
        "favourites_count": 5466,
        "utc_offset": null,
        "time_zone": null,
        "geo_enabled": true,
        "verified": true,
        "statuses_count": 45388,
        "lang": null,
        "contributors_enabled": false,
        "is_translator": false,
        "is_translation_enabled": false,
        "profile_background_color": "FFFFFF",
        "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
        "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
        "profile_background_tile": false,
        "profile_image_url": "http://pbs.twimg.com/profile_images/1345777059910586368/7Xq3mnJ8_normal.jpg",
        "profile_image_url_https": "https://pbs.twimg.com/profile_images/1345777059910586368/7Xq3mnJ8_normal.jpg",
        "profile_banner_url": "https://pbs.twimg.com/profile_banners/15207668/1609693230",
        "profile_link_color": "006699",
        "profile_sidebar_border_color": "FFFFFF",
        "profile_sidebar_fill_color": "FFFFFF",
        "profile_text_color": "000000",
        "profile_use_background_image": true,
        "has_extended_profile": false,
        "default_profile": false,
        "default_profile_image": false,
        "following": false,
        "follow_request_sent": false,
        "notifications": false,
        "translator_type": "none",
        "withheld_in_countries": []
    },
    "geo": null,
    "coordinates": null,
    "place": null,
    "contributors": null,
    "is_quote_status": false,
    "retweet_count": 167,
    "favorite_count": 476,
    "favorited": false,
    "retweeted": false,
    "possibly_sensitive": false,
    "lang": "en"
}]
```

**Comment by 1nVitr0** [comment]
The issue here is probably, that your json is not a collection of simple key-value pairs.

The extension currently uses a simplified json parser, that only supports arraus of single level json objects.

I might switch to parsing the actual json. With that I can either filter out object values, or maybe even supper dot-paths.

**Comment by 1nVitr0** [comment]
@otichy I extended the JSON support to multi-level objects by flattening them (with a `.` separator). This extension is definitely only for converting to and from flat list. It will never support more complicated objects, as there are too many language specific quirks. Also, there are way better and more efficient extensions out there that can address those specific needs.

With the latest update the extension now supports deep json objects by flattening. It also supports keyed json objects such as:

```json
{
    "thumb": {
        "w": 150,
        "h": 150,
        "resize": "crop"
    },
    "medium": {
        "w": 1200,
        "h": 675,
        "resize": "fit"
    },
    "small": {
        "w": 680,
        "h": 383,
        "resize": "fit"
    },
    "large": {
        "w": 1280,
        "h": 720,
        "resize": "fit"
    }
}
```

Though, as your above example only contains a single object it might be not that useful. You can always run the command with only the relevant parts selected.

**Comment by otichy** [comment]
Thanks for the enhancement. Flattening is a great idea. However, the JSON above does not produce a valid XML - some of the tags are not closed for some reasons. What I get is:

```
<root>
    <item>
        <created_at>Thu Sep 01 10:27:00 +0000 2022</created_at>
        <id>1565285083732496400</id>
        <id_str>1565285083732496384</id_str>
        <full_text>WATCH 📺@RepWaltzPress slams Biden’s desperate effort to change the narrative on Democrats’ ‘Defund the Police’ movement and America’s crime crisis:

“I can point to actions all day long that the Left has taken to try and defund our police at every level.” https://t.co/8zjf3WnHt6</full_text>
        <truncated>false
        <display_text_range.0>0</display_text_range.0>
        <display_text_range.1>255</display_text_range.1>
        <entities.hashtags>null
        <entities.symbols>null
        <entities.user_mentions.0.screen_name>RepWaltzPress</entities.user_mentions.0.screen_name>
        <entities.user_mentions.0.name>Congressman Waltz Press</entities.user_mentions.0.name>
        <entities.user_mentions.0.id>1082416697380913200</entities.user_mentions.0.id>
        <entities.user_mentions.0.id_str>1082416697380913152</entities.user_mentions.0.id_str>
        <entities.user_mentions.0.indices.0>7</entities.user_mentions.0.indices.0>
        <entities.user_mentions.0.indices.1>21</entities.user_mentions.0.indices.1>
        <entities.urls>null
        <entities.media.0.id>1564980601953943600</entities.media.0.id>
        <entities.media.0.id_str>1564980601953943552</entities.media.0.id_str>
        <entities.media.0.indices.0>256</entities.media.0.indices.0>
        <entities.media.0.indices.1>279</entities.media.0.indices.1>
        <entities.media.0.media_url>http://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg</entities.media.0.media_url>
        <entities.media.0.media_url_https>https://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg</entities.media.0.media_url_https>
        <entities.media.0.url>https://t.co/8zjf3WnHt6</entities.media.0.url>
        <entities.media.0.display_url>pic.twitter.com/8zjf3WnHt6</entities.media.0.display_url>
        <entities.media.0.expanded_url>https://twitter.com/HouseGOP/status/1565285083732496384/video/1</entities.media.0.expanded_url>
        <entities.media.0.type>photo</entities.media.0.type>
        <entities.media.0.sizes.thumb.w>150</entities.media.0.sizes.thumb.w>
        <entities.media.0.sizes.thumb.h>150</entities.media.0.sizes.thumb.h>
        <entities.media.0.sizes.thumb.resize>crop</entities.media.0.sizes.thumb.resize>
        <entities.media.0.sizes.medium.w>1200</entities.media.0.sizes.medium.w>
        <entities.media.0.sizes.medium.h>675</entities.media.0.sizes.medium.h>
        <entities.media.0.sizes.medium.resize>fit</entities.media.0.sizes.medium.resize>
        <entities.media.0.sizes.small.w>680</entities.media.0.sizes.small.w>
        <entities.media.0.sizes.small.h>383</entities.media.0.sizes.small.h>
        <entities.media.0.sizes.small.resize>fit</entities.media.0.sizes.small.resize>
        <entities.media.0.sizes.large.w>1280</entities.media.0.sizes.large.w>
        <entities.media.0.sizes.large.h>720</entities.media.0.sizes.large.h>
        <entities.media.0.sizes.large.resize>fit</entities.media.0.sizes.large.resize>
        <extended_entities.media.0.id>1564980601953943600</extended_entities.media.0.id>
        <extended_entities.media.0.id_str>1564980601953943552</extended_entities.media.0.id_str>
        <extended_entities.media.0.indices.0>256</extended_entities.media.0.indices.0>
        <extended_entities.media.0.indices.1>279</extended_entities.media.0.indices.1>
        <extended_entities.media.0.media_url>http://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg</extended_entities.media.0.media_url>
        <extended_entities.media.0.media_url_https>https://pbs.twimg.com/amplify_video_thumb/1564980601953943552/img/XipPz9MMK7F2sP9L.jpg</extended_entities.media.0.media_url_https>
        <extended_entities.media.0.url>https://t.co/8zjf3WnHt6</extended_entities.media.0.url>
        <extended_entities.media.0.display_url>pic.twitter.com/8zjf3WnHt6</extended_entities.media.0.display_url>
        <extended_entities.media.0.expanded_url>https://twitter.com/HouseGOP/status/1565285083732496384/video/1</extended_entities.media.0.expanded_url>
        <extended_entities.media.0.type>video</extended_entities.media.0.type>
        <extended_entities.media.0.sizes.thumb.w>150</extended_entities.media.0.sizes.thumb.w>
        <extended_entities.media.0.sizes.thumb.h>150</extended_entities.media.0.sizes.thumb.h>
        <extended_entities.media.0.sizes.thumb.resize>crop</extended_entities.media.0.sizes.thumb.resize>
        <extended_entities.media.0.sizes.medium.w>1200</extended_entities.media.0.sizes.medium.w>
        <extended_entities.media.0.sizes.medium.h>675</extended_entities.media.0.sizes.medium.h>
        <extended_entities.media.0.sizes.medium.resize>fit</extended_entities.media.0.sizes.medium.resize>
        <extended_entities.media.0.sizes.small.w>680</extended_entities.media.0.sizes.small.w>
        <extended_entities.media.0.sizes.small.h>383</extended_entities.media.0.sizes.small.h>
        <extended_entities.media.0.sizes.small.resize>fit</extended_entities.media.0.sizes.small.resize>
        <extended_entities.media.0.sizes.large.w>1280</extended_entities.media.0.sizes.large.w>
        <extended_entities.media.0.sizes.large.h>720</extended_entities.media.0.sizes.large.h>
        <extended_entities.media.0.sizes.large.resize>fit</extended_entities.media.0.sizes.large.resize>
        <extended_entities.media.0.video_info.aspect_ratio.0>16</extended_entities.media.0.video_info.aspect_ratio.0>
        <extended_entities.media.0.video_info.aspect_ratio.1>9</extended_entities.media.0.video_info.aspect_ratio.1>
        <extended_entities.media.0.video_info.duration_millis>134401</extended_entities.media.0.video_info.duration_millis>
        <extended_entities.media.0.video_info.variants.0.bitrate>832000</extended_entities.media.0.video_info.variants.0.bitrate>
        <extended_entities.media.0.video_info.variants.0.content_type>video/mp4</extended_entities.media.0.video_info.variants.0.content_type>
        <extended_entities.media.0.video_info.variants.0.url>https://video.twimg.com/amplify_video/1564980601953943552/vid/640x360/pM47QMwLx9_eCrDB.mp4?tag=14</extended_entities.media.0.video_info.variants.0.url>
        <extended_entities.media.0.video_info.variants.1.bitrate>2176000</extended_entities.media.0.video_info.variants.1.bitrate>
        <extended_entities.media.0.video_info.variants.1.content_type>video/mp4</extended_entities.media.0.video_info.variants.1.content_type>
        <extended_entities.media.0.video_info.variants.1.url>https://video.twimg.com/amplify_video/1564980601953943552/vid/1280x720/OpWFIA3y2UnlWJy5.mp4?tag=14</extended_entities.media.0.video_info.variants.1.url>
        <extended_entities.media.0.video_info.variants.2.bitrate>288000</extended_entities.media.0.video_info.variants.2.bitrate>
        <extended_entities.media.0.video_info.variants.2.content_type>video/mp4</extended_entities.media.0.video_info.variants.2.content_type>
        <extended_entities.media.0.video_info.variants.2.url>https://video.twimg.com/amplify_video/1564980601953943552/vid/480x270/_61qh2kjP6iPER7v.mp4?tag=14</extended_entities.media.0.video_info.variants.2.url>
        <extended_entities.media.0.video_info.variants.3.content_type>application/x-mpegURL</extended_entities.media.0.video_info.variants.3.content_type>
        <extended_entities.media.0.video_info.variants.3.url>https://video.twimg.com/amplify_video/1564980601953943552/pl/3XiKmwB_IDNPVHqV.m3u8?tag=14&container=fmp4</extended_entities.media.0.video_info.variants.3.url>
        <extended_entities.media.0.additional_media_info.title>Rep. Michael Waltz on Biden's 'Defund the Police' blame game</extended_entities.media.0.additional_media_info.title>
        <extended_entities.media.0.additional_media_info.description></extended_entities.media.0.additional_media_info.description>
        <extended_entities.media.0.additional_media_info.call_to_actions.visit_site.url>https://www.gop.gov/</extended_entities.media.0.additional_media_info.call_to_actions.visit_site.url>
        <extended_entities.media.0.additional_media_info.embeddable>true
        <extended_entities.media.0.additional_media_info.monetizable>false
        <source><a href="https://studio.twitter.com" rel="nofollow">Twitter Media Studio</a></source>
        <in_reply_to_status_id>null
        <in_reply_to_status_id_str>null
        <in_reply_to_user_id>null
        <in_reply_to_user_id_str>null
        <in_reply_to_screen_name>null
        <user.id>15207668</user.id>
        <user.id_str>15207668</user.id_str>
        <user.name>House Republicans</user.name>
        <user.screen_name>HouseGOP</user.screen_name>
        <user.location>Washington, DC</user.location>
        <user.description>We are the Republican Conference in the United States House of Representatives 🇺🇸</user.description>
        <user.url>https://t.co/lcTnaxp5EE</user.url>
        <user.entities.url.urls.0.url>https://t.co/lcTnaxp5EE</user.entities.url.urls.0.url>
        <user.entities.url.urls.0.expanded_url>http://gop.gov</user.entities.url.urls.0.expanded_url>
        <user.entities.url.urls.0.display_url>gop.gov</user.entities.url.urls.0.display_url>
        <user.entities.url.urls.0.indices.0>0</user.entities.url.urls.0.indices.0>
        <user.entities.url.urls.0.indices.1>23</user.entities.url.urls.0.indices.1>
        <user.entities.description.urls>null
        <user.protected>false
        <user.followers_count>1358518</user.followers_count>
        <user.friends_count>798</user.friends_count>
        <user.listed_count>7894</user.listed_count>
        <user.created_at>Mon Jun 23 14:41:52 +0000 2008</user.created_at>
        <user.favourites_count>5466</user.favourites_count>
        <user.utc_offset>null
        <user.time_zone>null
        <user.geo_enabled>true
        <user.verified>true
        <user.statuses_count>45388</user.statuses_count>
        <user.lang>null
        <user.contributors_enabled>false
        <user.is_translator>false
        <user.is_translation_enabled>false
        <user.profile_background_color>FFFFFF</user.profile_background_color>
        <user.profile_background_image_url>http://abs.twimg.com/images/themes/theme1/bg.png</user.profile_background_image_url>
        <user.profile_background_image_url_https>https://abs.twimg.com/images/themes/theme1/bg.png</user.profile_background_image_url_https>
        <user.profile_background_tile>false
        <user.profile_image_url>http://pbs.twimg.com/profile_images/1345777059910586368/7Xq3mnJ8_normal.jpg</user.profile_image_url>
        <user.profile_image_url_https>https://pbs.twimg.com/profile_images/1345777059910586368/7Xq3mnJ8_normal.jpg</user.profile_image_url_https>
        <user.profile_banner_url>https://pbs.twimg.com/profile_banners/15207668/1609693230</user.profile_banner_url>
        <user.profile_link_color>006699</user.profile_link_color>
        <user.profile_sidebar_border_color>FFFFFF</user.profile_sidebar_border_color>
        <user.profile_sidebar_fill_color>FFFFFF</user.profile_sidebar_fill_color>
        <user.profile_text_color>000000</user.profile_text_color>
        <user.profile_use_background_image>true
        <user.has_extended_profile>false
        <user.default_profile>false
        <user.default_profile_image>false
        <user.following>false
        <user.follow_request_sent>false
        <user.notifications>false
        <user.translator_type>none</user.translator_type>
        <user.withheld_in_countries>null
        <geo>null
        <coordinates>null
        <place>null
        <contributors>null
        <is_quote_status>false
        <retweet_count>167</retweet_count>
        <favorite_count>476</favorite_count>
        <favorited>false
        <retweeted>false
        <possibly_sensitive>false
        <lang>en</lang>
    </item>
</root>
```

**Comment by 1nVitr0** [comment]
Looks like it might be time to add some automated tests to the extension... I'm sure that's a quick fix for now, but tests would've shown the issue already.

**Comment by 1nVitr0** [comment]
Should be fixed by #9

**Comment by otichy** [comment]
Thanks for also solving the entity conversion. 🙂
